Sources/DangerDependenciesResolver/Script.swift

```diff
@@ -0,0 +1,299 @@
+
+        return editingPath()
+    }
+
+    private func editingPath() -> String {
+        folder.appendingPath(name + ".xcodeproj")
+    }
+
+    private func generateXCodeProjWithConfig(configPath: String) throws {
+        try executeSwiftCommand("package generate-xcodeproj --xcconfig-overrides \(configPath)", onFolder: folder, executor: ShellExecutor())
+    }
+
+    private func sourcesImportPath(forImportPath importPath: String) -> String {
+        folder
+            .appendingPath("Sources")
+            .appendingPath(name)
+            .appendingPath(importPath.fileName)
+    }
+
+    public func watch(importedFiles: [String]) throws {
+        let fullPathImports = importedFiles.map { $0.fullPath }
+        try watch(imports: fullPathImports)
+        try? copyImports(fullPathImports)
+    }
+
+    public func watch(imports: [String]) throws {
+        do {
+            let path = editingPath()
+
+            try ShellExecutor().spawn("open \"\(path)\"", arguments: [])
+
+            logger.logInfo("\nℹ️  Danger will keep running, in order to commit any changes you make in Xcode back to the original script file")
+            logger.logInfo("   Press the return key once you're done")
+
+            startCopyLoop(imports: imports)
+            _ = FileHandle.standardInput.availableData
+            try copyChangesToSymlinkedFile()
+        } catch {
+            throw Errors.watchingFailed(name)
+        }
+    }
+
+    private func startCopyLoop(imports: [String]) {
+        let dispatchQueue: DispatchQueue
+
+        if let existingQueue = copyLoopDispatchQueue {
+            dispatchQueue = existingQueue
+        } else {
+            let newQueue = DispatchQueue(label: "com.danger.fileCopyLoop")
+            copyLoopDispatchQueue = newQueue
+            dispatchQueue = newQueue
+        }
+
+        dispatchQueue.asyncAfter(deadline: .now() + .seconds(3)) { [weak self] in
+            try? self?.copyChangesToSymlinkedFile()
+            try? self?.copyImports(imports)
+            self?.startCopyLoop(imports: imports)
+        }
+    }
+
+    private func copyChangesToSymlinkedFile() throws {
+        let script = try expandSymlink()
+
+        let data = try Data(contentsOf: URL(fileURLWithPath: folder.appendingPath(localPath)))
+        try data.write(to: URL(fileURLWithPath: script))
+    }
+
+    private func expandSymlink() throws -> String {
+        try ShellExecutor().spawn("readlink \(folder.appendingPath("OriginalFile"))", arguments: [])
+    }
+
+    private func copyImports(_ imports: [String]) throws {
+        try imports.forEach { importPath in
+            try Data(contentsOf: URL(fileURLWithPath: sourcesImportPath(forImportPath: importPath))).write(to: URL(fileURLWithPath: importPath))
+        }
+    }
+}
+
+@discardableResult
+func executeSwiftCommand(_ command: String, onFolder folder: String? = nil, arguments: [String] = [], executor: ShellExecutor) throws -> String {
```

Comment:
<!--
  0 failure: 
  1 warning:  Line should be 12...
  
  
  DangerID: danger-id-Danger;
  File: Sources/DangerDependenciesResolver/Script.swift;
  Line: 235;
-->

- :warning: Line should be 120 characters or less: currently 145 characters (`line_length`)